core/generator/generator.go

```diff
@@ -75,7 +75,16 @@ func Generate(
 		log.Fatal(ctx, log.KeyError, err)
 	}
 	if b != nil && (g.latestBlock == nil || b.Height == g.latestBlock.Height+1) {
-		s := state.Copy(g.latestSnapshot)
+		var s *state.Snapshot
+		if g.latestSnapshot == nil {
```

Comment:
> so maybe Chain.Recover can just error out in the empty-blockchain case

I think we need to support that case for booting a new participant node w/o a snapshot. Can we pass the blockchain ID into `Recover` instead?

> I also made CommitBlock synchronously store a snapshot for the initial block (because I was getting non-deterministic failures in TestGetAndAddBlockSignatures)

Hrm, can we make any changes to TestGetAndAddBlockSignatures to make it more reliable? It'd be nice to avoid special cases like this because it makes the invariants more complicated.